DirectProgramming/DPC++/StructuredGrids/iso2dfd_dpcpp/README.md

```diff
@@ -0,0 +1,140 @@
+
+SYCL implementation explained.  
+
+* DPC++ queues (including device selectors and exception handlers).
+* DPC++ buffers and accessors.  
+* The ability to call a function inside a kernel definition and pass accessor arguments as pointers. A 
+function called inside the kernel performs a computation (it updates a grid point specified by the 
+global ID variable) for a single time step.  
+
+
+## License
+
+This code sample is licensed under MIT license.  
+
+
+##  Building the `iso2dfd` Program for CPU and GPU
+
+### Running Samples In DevCloud
+
+If running a sample in the Intel DevCloud, remember that you must specify the compute node (CPU, GPU, 
+FPGA) as well whether to run in batch or interactive mode. For more information see the Intel® oneAPI 
+Base Toolkit Get Started Guide (https://devcloud.intel.com/oneapi/get-started/base-toolkit/)
+
+### On a Linux* System
+Perform the following steps:
+1. Build the program using the following `cmake` commands. 
+
+  ```    
+    cd iso2dfd_dpcpp &&  
+    mkdir build &&  
+    cd build &&  
+    cmake .. &&  
+    make -j 
+    ```
+
+2. Run the program on Gen9 
+
+    ```
+    make run  
+    ```
+
+3. Clean the program  
+  
+    ```
+    make clean  
+    ```
+
+
+### On a Windows* System Using Visual Studio* Version 2017 or Newer
+* Build the program using VS2017 or VS2019
+      Right click on the solution file and open using either VS2017 or VS2019 IDE.
+      Right click on the project in Solution explorer and select Rebuild.
+      From top menu select Debug -> Start without Debugging.
+
+* Build the program using MSBuild
+      Open "x64 Native Tools Command Prompt for VS2017" or "x64 Native Tools Command Prompt for VS2019"
+      Run - MSBuild iso2dfd.sln /t:Rebuild /p:Configuration="Release"
+
+
+## Running the Sample
+
+### Application Parameters 
+
+You can  execute the code with different parameters. For example the following command will run the iso2dfd executable using a 1000x1000 grid size and it will iterate over 2000 time steps.
+
+    ```
+    ./iso2dfd 1000 1000 2000
+    ```	
+ 	
+   Usage: ./iso2dfd n1 n2 Iterations
+
+	 n1 n2      : Grid sizes for the stencil
+	 Iterations : Number of timesteps.
+
+   * Find graphical output for sample execution in the online tutorial at:
+     https://software.intel.com/en-us/articles/code-sample-two-dimensional-finite-difference-wave-propagation-in-isotropic-media-iso2dfd
+
+### Example of Output 
+
+    ```
+    Initializing ... 
+    Grid Sizes: 1000 1000
+    Iterations: 2000
+
+    Computing wavefield in device ..
+    Running on Intel(R) Gen9 HD Graphics NEO
+    The Device Max Work Group Size is : 256
+    The Device Max EUCount is : 24
+    SYCL time: 3282 ms
+
+    Computing wavefield in CPU ..
+    Initializing ... 
+    CPU time: 8846 ms
+
+    Final wavefields from device and CPU are equivalent: Success
+    Final wavefields (from device and CPU) written to disk
+    Finished.  
+    [100%] Built target run
+    ```
```

Comment:
The readme needs to have a statement about 'common' file and where to find them.

Reply:
Done. 
Fixed copyright year and added statement about 'common' file to README file
Thanks .